engine/runtime/src/main/java/org/enso/interpreter/instrument/IdExecutionInstrument.java

```diff
@@ -166,7 +188,10 @@ public void onReturnValue(EventContext context, VirtualFrame frame, Object resul
                 (FunctionCallInstrumentationNode.FunctionCall) result));
       } else if (node instanceof ExpressionNode) {
         valueCallback.accept(
-            new ExpressionValue(((ExpressionNode) context.getInstrumentedNode()).getId(), result));
+            new ExpressionValue(
+                ((ExpressionNode) context.getInstrumentedNode()).getId(),
```

Comment:
This should reuse `node` to make it clearer why the cast is safe. 

```suggestion
                ((ExpressionNode) node).getId(),
```